service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobClientOptions.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.sdk.iot.service.jobs;
+
+import com.microsoft.azure.sdk.iot.service.ProxyOptions;
+import lombok.Builder;
+import lombok.Getter;
+
+/**
+ * Configurable options for all job client operations
+ */
+@Builder
+public class JobClientOptions
+{
+    protected static final Integer DEFAULT_HTTP_READ_TIMEOUT_MS = 24000; // 24 seconds
+    protected static final Integer DEFAULT_HTTP_CONNECT_TIMEOUT_MS = 24000; // 24 seconds
+
+    /**
+     * The options that specify what proxy to tunnel through. If null, no proxy will be used
+     */
+    @Getter
+    private final ProxyOptions proxyOptions;
+
+    /**
+     * The http read timeout to a specified timeout, in milliseconds. A non-zero value specifies the timeout when reading from
+     * Input stream after a connection is established to a resource. If the timeout expires before there is data available
+     * for read, a java.net.SocketTimeoutException is raised. A timeout of zero is interpreted as an infinite timeout.
+     * By default, this value is {@link #DEFAULT_HTTP_READ_TIMEOUT_MS}. Must be a non-negative value.
+     */
+    @Getter
+    private final int httpReadTimeout;
```

Comment:
Do we have the option of using `Duration` or `Period` instead? I am not sure if they are supported in Java 7 though ... I like using Timespan in .NET instead of having an integer with no unit indication ... I am just thinking out loud

Reply:
Thankfully the service client has always been Java 8+. Despite that, the HTTP library that we use takes these values as ints, so it feels appropriate for us to ask for them as ints in order to avoid any conversion problems.